apps/cloud/src/application/blob_service.rs

```diff
@@ -0,0 +1,545 @@
+use crate::{
+    context::Context, infrastructure::auth::get_claim_from_headers,
+    infrastructure::error_status::ErrorStatus,
+};
+use axum::{
+    extract::BodyStream,
+    headers::ContentLength,
+    http::{
+        header::{
+            CACHE_CONTROL, CONTENT_LENGTH, CONTENT_TYPE, ETAG, IF_MODIFIED_SINCE, IF_NONE_MATCH,
+            LAST_MODIFIED,
+        },
+        HeaderMap, HeaderValue, Method,
+    },
+    Json,
+};
+use chrono::{DateTime, Utc};
+use futures::{future, StreamExt};
+use jwst::error;
+use jwst::BlobStorage;
+use jwst_logger::{info, instrument, tracing};
+use mime::APPLICATION_OCTET_STREAM;
+use serde::Serialize;
+use std::sync::Arc;
+use std::{collections::HashMap, path::PathBuf};
+
+#[derive(Serialize)]
+pub struct Usage {
+    pub blob_usage: BlobUsage,
+}
+
+#[derive(Serialize)]
+pub struct BlobUsage {
+    pub usage: u64,
+    pub max_usage: u64,
+}
+
+const MAX_USAGE: u64 = 10 * 1024 * 1024 * 1024;
+const MAX_BLOB_SIZE: u64 = 10 * 1024 * 1024;
+
+#[derive(Debug)]
+pub struct BlobService {}
+
+impl BlobService {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    #[instrument(skip(ctx, method, headers))]
+    pub async fn get_blob_in_workspace(
+        &self,
+        ctx: &Arc<Context>,
+        workspace_id: String,
+        id: String,
+        method: Method,
+        headers: HeaderMap,
+    ) -> Result<(HeaderMap, Option<Vec<u8>>), ErrorStatus> {
+        info!("get_blob_in_workspace enter");
```

Comment:
add id or something related into the log message

Reply:
Code like "#[instrument(skip(ctx, method, headers))]" has already carried the values of the function's arguments.